controllers/workspace/eventhandlers.go

```diff
@@ -47,44 +47,63 @@ func dwRelatedPodsHandler(obj client.Object) []reconcile.Request {
 }
 
 func (r *DevWorkspaceReconciler) dwPVCHandler(obj client.Object) []reconcile.Request {
-	// Check if PVC is owned by a DevWorkspace (per-workspace storage case)
-	for _, ownerref := range obj.GetOwnerReferences() {
-		if ownerref.Kind != "DevWorkspace" {
-			continue
-		}
-		return []reconcile.Request{
-			{
-				NamespacedName: types.NamespacedName{
-					Name:      ownerref.Name,
-					Namespace: obj.GetNamespace(),
+	// No need to reconcile PVC, if it doesn't have a label specifying its type
+	pvcLabel, ok := obj.GetLabels()[constants.DevWorkspacePVCTypeLabel]
```

Comment:
If a user upgrades from an older version of DWO to the latest version, will the older common PVCs get the new label?

Reply:
@mkuznyetsov this is also my concern. There was an older [PR](https://github.com/devfile/devworkspace-operator/pull/1174) that added labels per-workspace PVCs, but I'm almost certain this only impacted _new_ per-workspace PVCs.

My concern stems from 2 things:

## 1. Making sure existing PVCs won't be deleted, then re-created to apply the label.
This would cause user data loss. Looking at the sync code, it seems like PVCs are [not considered mutable](https://github.com/devfile/devworkspace-operator/blob/main/pkg/provision/sync/sync.go#L183), so the new label would not get applied to existing PVCs. There is however a note that we could still allow for [updating labels](https://github.com/devfile/devworkspace-operator/blob/main/pkg/provision/sync/sync.go#L59).

If we want to update existing PVCs with the label, this complicates this PR further: either the sync code needs to be updated to only allow modifying PVCs if the labels are modified, or we need to manually `diff` the PVC after calling `SyncObjectWithCluster()`, see that the labels are missing and call` sync.updateObjectGeneric()` to apply the labels. The former approach is much more clean, though a bit trickier as `SyncObjectWithCluster()` is a critical function called throughout DWO. 

## 2. If we decide to not update existing PVCs with the new label, then the `dwPVCHandler()` should still work on PVCs that are missing the new label. 
This is currently not the case with the following check:

```go
	// No need to reconcile, if PVC doesn't have a label specifying its type
	pvcLabel, ok := obj.GetLabels()[constants.DevWorkspacePVCTypeLabel]

	if !ok {
		return []reconcile.Request{}
	}
```

You'd have to change the `dwPVCHandler` code to not assume that the `constants.DevWorkspacePVCTypeLabel` is applied to the PVC in question. If it does not exist, the behaviour should match the behaviour prior to this PR (i.e. the global common PVC name is assumed to be used, and the original PR bug will be present). If the label does exist, then you can make use of the logic that you've implemented. 

---

In general, I think it's easier as a first step to have this PR _not_ update existing PVCs with the new label (i.e. consider my 2nd concern). A later PR could update the sync code to allow label updating on PVCs.

Reply:
alright, I will rewrite to account for the use of PVC with missing label, and leave a reference to this PR why this has to be this way

Reply:
pr rewritten to support old usecase, when there is no new label